src_c/polygon.c

```diff
@@ -468,6 +468,34 @@ pg_polygon_move(pgPolygonObject *self, PyObject *const *args, Py_ssize_t nargs)
     return tmp;
 }
 
+static PyObject *
+pg_polygon_as_segments(pgPolygonObject *self, PyObject *_null) {
+    double *verts = self->polygon.vertices;
+    Py_ssize_t vert_num = self->polygon.verts_num;
+    PyObject *list = PyList_New(vert_num);
+    for (Py_ssize_t i = 0; i < vert_num * 2; i+=2) {
+        double next_vertex_x;
+        double next_vertex_y;
+
+        if (i >= vert_num * 2 - 2) {
+            next_vertex_x = verts[0];
+            next_vertex_y = verts[1];
+        }
+        else {
+            next_vertex_x = verts[i + 2];
+            next_vertex_y = verts[i + 3];
+        }
+
+        PyObject *line = pgLine_New4(verts[i], verts[i + 1], next_vertex_x, next_vertex_y);
+        if (!line) {
+            return NULL;
+        }
+        PyList_SET_ITEM(list, i / 2, line);
+    }
```

Comment:
To be honest you could make this much simpler by doing this:
```suggestion
    for (i = 0; i < verts_num; i++) {
        Py_ssize_t i2 = i * 2;
        PyObject *line = pgLine_New4(vertices[i2], vertices[i2 + 1],
                                        vertices[(i2 + 2) % (verts_num*2)],
                                        vertices[(i2 + 3) % (verts_num*2)]);
        if (!line) {
            Py_DECREF(list);
            return NULL;
        }
        PyList_SET_ITEM(list, i, line);
    }
```

Reply:
you could also put that verts_num*2 into a separate variable